### ISSUE TITLE
Clicking on ref link leads to wrong location

### ISSUE THREAD

**Issue by jstac** [najuzilu/sphinxcontrib-prettyproof#27]
Maybe "wrong location" is an exaggeration, but what I mean is that if you click on a link that says "Theorem 1" in Chrome, you are directed to the theorem body, while the header "Theorem 1" is not visible --- it's off the top of the screen.

Hence it's not clear that you have been directed to the right place unless you scroll up.

**Comment by najuzilu** [comment]
Yes, I have noticed that as well. It's actually because of the main topbar which has a white background and it's almost the same height as the title div of the directive.

**Comment by jstac** [comment]
Is there any way to fix it you can think of?

**Comment by mmcky** [comment]
@najuzilu would it help to wrap the whole `theorem` in another `div` and set the `id=target` to get link to the top of a `theorem`? @AakashGfude may be able to help on the `html` side of things here.

**Comment by AakashGfude** [comment]
I think the easiest way might be to control the scroll behaviour using javascript and provide a bit of an offset to the location scrolled. 
If @najuzilu is not working on this, I can give this one a shot.

**Comment by najuzilu** [comment]
I have a working solution for same-file targets, but I need to account for targets leading to different locations. I decided to go with JS and animate the scroll. How does that sound @AakashGfude?

**Comment by AakashGfude** [comment]
sounds good @najuzilu . I guess it will be a more generic solution.
